niap-cc/Permissions/Tester/app/src/main/java/com/android/certifications/niap/permissions/utils/ReflectionUtils.java

```diff
@@ -54,6 +65,103 @@ public static Object invokeReflectionCall(Class<?> targetClass, String methodNam
         }
     }
 
+
+    public static Object stubFromInterface(String className)  {
+        try {
+            Class<?> ifListner = Class.forName(className);
+            Object listenerInstance = Proxy.newProxyInstance(
+                    ifListner.getClassLoader(), new Class<?>[]{ifListner}, new InvocationHandler() {
+                @Override
+                public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
+                    //mLogger.logDebug("callback invoked: "+proxy+","+method);
+                    if (method.getName().equals("asBinder")) {
+                        return new Binder() ;
+                    }
+                    if(method.getName().equals("onSuccess")){
+                        mLogger.logDebug("onSuccess");
+                        //System.out.println("ARGS: " + (Integer)args[0] + ", " + (Integer)args[1]);
+                        return 1;
+                    } else if(method.getName().equals("onError")){
+                        mLogger.logDebug("onError");
+                        //throw new SecurityException("Errror!");
+                        return 1;
+                    }
+                    else return -1;
+                }
+            });
+            mLogger.logDebug(checkDeclaredMethod(ifListner,"").toString());;
+
+            return ifListner.cast(listenerInstance);
+        } catch (ClassNotFoundException e){
+            e.printStackTrace();
+            return null;
+        }
+    }
+    public static boolean deviceConfigSetProperty(String a,String b,String c,boolean value)  {
```

Comment:
Ultra-nit: in a follow-up pull request, can you rename the parameters `a`,  `b`, and `c`? I believe `a` and `b` are referring to the `name` and `namespace`, but I'm not sure to what `c` refers.

Reply:
Thanks , fix it.
Actually the 3rd argument is value, 4th parameter is makeDefault. 
As I checked callers correctly handle them.

[Android Source Code](https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/provider/DeviceConfig.java;drc=4f6338f7ab0946fcab04cc2554559be1befbba77;l=972)

Reply:
Thanks for the additional info, update looks good to me!